### DISCUSSION TITLE
Make setting the LESS environment var work better with git

### DISCUSSION THREAD

**Discussion by nicvagn** [ohmybash/oh-my-bash#539] [Ideas]
If you 
export LESS=-FRX

git does not pipe output (ie: git branch) through less unless it is more than a page.

I do not know where LESS is set as an environment var, but the setting as it is interferes wit my git workflow

**Comment by akinomyoga** [comment]
If you want to override the environment variable, you can write the setting after `source "$OSH"/oh-my-bash.sh` in `~/.bashrc`.

**Reply by nicvagn** [reply]
And I have, but what I was meaning was I think a re-examination of the defaults could be helpful. 

I just wanted to aid in making oh-my-bash's out of the box config more ideal, and prevent others from going down the rabbit hole I went down.

I got mine, but maybe I can help others?

**Reply by nicvagn** [reply]
If LESS is set, git uses it to pipe output. This is not ideal, as it does it when output is short. The -FRX flags only do so if output is long

**Reply by akinomyoga** [reply]
> And I have, but what I was meaning was I think a re-examination of the defaults could be helpful.

If proper examination and investigation are made to support the change, I can accept a PR. I'll have to request at least investigations on the background for the current default value and the impact of changing that including the possible breakage of other tools. Also, we need to care about the other use cases that may use the current default values. We shall compare them with the use cases that are useful with the suggested value `-FRX`.

You only mention the usage for `git`, but `LESS` is not a configuration variable for `git`, but a general setting for any tools that use the `less` command. Some tools assume that the terminal content becomes the original one after the `less` command ends. The suggested value breaks that assumption. How would you solve that issue?

Or maybe another option is just to unset `LESS`, so that each tool decides the option if needed.

**Reply by nicvagn** [reply]
That is a fair point. What is the reason fror setting the LESS environment variable? If it is just because the oh-my-zsh does it, why do they do it?

I can not check every tool and config. 

I just know how it is set, interferes with native git. un setting it may be a more flexible option

**Reply by nicvagn** [reply]
I just wanted to bring it up. I do not even know where to start on said investigation. 

But git is a widely used tool, and as it is set right now, interferes with it. 

There is a reason I did not make a pr. I made a discussion. Because I do not feel qualified to make config decision's for this project.

removing the setting may be best, IDK

**Reply by akinomyoga** [reply]
> That is a fair point. What is the reason fror setting the LESS environment variable?

I'm not sure because I just took over the maintenance so do not know every detail of the background of each setting.

> If it is just because the oh-my-zsh does it, why do they do it?

If it is just imported from Oh My Zsh, I think we can just look at what the upstream Oh My Zsh does with it. If they already changed the default value to be the one that works better with `git`, we can just follow it after checking the relevant discussion in the upstream. Or, if they removed it, we can remove it similarly.

> I can not check every tool and config.

Yes, it's impossible to make a perfect survey. However, we should make a certain level of effort before making a possibly breaking change. At least, I cannot merge a breaking change without any investigations or considerations about possible impact.

> I just wanted to bring it up. I do not even know where to start on said investigation.

I'd appreciate it if you could check how Oh My Zsh handles it. Maybe OMZ doesn't set the `LESS` environment, or OMZ sets a certain value. If it is different from the setup in OMB, we'd like to also check the change history in OMZ.

**Reply by nicvagn** [reply]
https://github.com/ohmyzsh/ohmyzsh/blob/fd01fd66ce27c669e5ffaea94460a37423d1e134/lib/misc.zsh#L22

here is where it is set.

From the man pages:

     R  -    Repaint the screen, discarding any buffered input.  That
               is, reload the current file.  Useful if the file is
               changing while it is being viewed.

so oh-my-zsh set's it, I will ask why

**Reply by akinomyoga** [reply]
> https://github.com/ohmyzsh/ohmyzsh/blob/fd01fd66ce27c669e5ffaea94460a37423d1e134/lib/misc.zsh#L22
> 
> here is where it is set.

Thank you for checking! The setup of OMZ seems to be the same as ours.

> so oh-my-zsh set's it, I will ask why

Yeah, I assume they know about it better. Let's see what they think about it.

**Reply by nicvagn** [reply]
Good call. No response yet.

**Reply by nicvagn** [reply]
In my own case, with some further use I find that export LESS=-FX is more usable. The R flag causes the screen to redraw, witch I do not personally want. 

I think the native configuration messing with a ubiquitous command like git is a bug.
:. it is more sane to leave it as it is, rather than introducing a bug by setting it. I will have to see what the justification was though.
